### PR TITLE
Fix reply of VFIO_USER_DEVICE_GET_REGION_INFO

### PR DESCRIPTION
**Pull request by swapnili**
Set VFIO_REGION_INFO_FLAG_CAPS flag only if caps are part of the reply.

Signed-off-by: Swapnil Ingle <swapnil.ingle@nutanix.com>

### CODE REVIEW

File: test/py/test_device_get_region_info.py [hunk 1]
```diff
@@ -136,8 +136,7 @@ def test_device_get_region_info_small_argsz_caps():
     assert info.argsz == 80
     assert info.flags == (VFIO_REGION_INFO_FLAG_READ |
```
Comment (jlevon) [inline test/py/test_device_get_region_info.py:137]: can we get a comment here? i.e. there are caps, but they don't fit, so we don't expect the flag to be set.

**Comment by jlevon** [comment]
is this the issue changpeng had?

**Comment by swapnili** [comment]
> is this the issue changpeng had?

Yes I think so this should solve that

**Review by jlevon — APPROVED** [review]
thank you

**Pull request merged**